contracts/modules/licensing/LicensingModuleV2.sol

```diff
@@ -0,0 +1,360 @@
+            if (i == 0) {
+                commonRoyaltyPolicy = royaltyPolicy;
+            } else if (royaltyPolicy != commonRoyaltyPolicy) {
+                revert Errors.LicensingModule__IncompatibleRoyaltyPolicy(royaltyPolicy, commonRoyaltyPolicy);
+            }
+        }
+
+        if (commonRoyaltyPolicy != address(0)) {
+            ROYALTY_MODULE.onLinkToParents(
+                derivativeIpId,
+                commonRoyaltyPolicy,
+                originalIpIds,
+                royaltyDatas,
+                royaltyContext
+            );
+        }
+        // burn license tokens
+        LICENSE_NFT.burnLicenseTokens(derivativeIpOwner, licenseTokenIds);
+        emit DerivativeRegisteredWithLicenseTokens(
+            msg.sender,
+            derivativeIpId,
+            licenseTokenIds,
+            originalIpIds,
+            licenseTermsIds,
+            licenseTemplate
+        );
+    }
+
+    function _payMintingFeeForAll(
+        address[] calldata originalIpIds,
+        uint256[] calldata licenseTermsIds,
+        address licenseTemplate,
+        address derivativeIpOwner,
+        bytes calldata royaltyContext
+    ) private returns (address commonRoyaltyPolicy, bytes[] memory royaltyDatas) {
+        commonRoyaltyPolicy = address(0);
+        royaltyDatas = new bytes[](licenseTermsIds.length);
+
+        for (uint256 i = 0; i < originalIpIds.length; i++) {
+            uint256 lcId = licenseTermsIds[i];
+            Licensing.MintingLicenseConfig memory mlc = LICENSE_REGISTRY.getMintingLicenseConfig(
+                originalIpIds[i],
+                licenseTemplate,
+                lcId
+            );
+            if (mlc.receiverCheckModule != address(0)) {
+                if (!IHookModule(mlc.receiverCheckModule).verify(derivativeIpOwner, mlc.receiverCheckData)) {
+                    revert Errors.LicensingModule__ReceiverCheckFailed(derivativeIpOwner);
+                }
+            }
+            (address royaltyPolicy, bytes memory royaltyData) = _payMintingFee(
+                originalIpIds[i],
+                licenseTemplate,
+                lcId,
+                1,
+                royaltyContext,
+                mlc
+            );
+            royaltyDatas[i] = royaltyData;
+
+            if (i == 0) {
+                commonRoyaltyPolicy = royaltyPolicy;
+            } else if (royaltyPolicy != commonRoyaltyPolicy) {
+                revert Errors.LicensingModule__IncompatibleRoyaltyPolicy(royaltyPolicy, commonRoyaltyPolicy);
+            }
+        }
+    }
+
+    function _payMintingFee(
+        address originalIpId,
+        address licenseTemplate,
+        uint256 licenseTermsId,
+        uint256 amount,
+        bytes calldata royaltyContext,
+        Licensing.MintingLicenseConfig memory mlc
+    ) private returns (address royaltyPolicy, bytes memory royaltyData) {
+        ILicenseTemplate lct = ILicenseTemplate(licenseTemplate);
+        uint256 mintingFee = 0;
```

Comment:
(Minor)
uint256 mintingFee = 0;
address currencyToken = address(0);

these 2 lines may not be needed

Reply:
Yeah, as we defined named returns `returns (address royaltyPolicy, bytes memory royaltyData)` , we have to define the `mintingFee` and `currencyToken` as well.